featurex/converters/video.py

```diff
@@ -0,0 +1,79 @@
+from featurex.stimuli.video import VideoStim, DerivedVideoStim, VideoFrameStim
+from featurex.converters import Converter
+
+import pandas as pd
+
+
+class VideoToDerivedVideoConverter(Converter):
+
+    ''' Base AudioToText Converter class; all subclasses can only be applied to
+    audio and convert to text. '''
+    target = VideoStim
+    _output_type = DerivedVideoStim
+
+
+class FrameSamplingConverter(VideoToDerivedVideoConverter):
+    ''' 
+    Samples frames from video stimuli, to improve efficiency
+
+    Args:
+        every (int): takes every nth frame
+        hertz (int): takes n frames per second
+        num_frames (int): takes top n frames sorted by the absolute difference
+         with the next frame
+    '''
+    def __init__(self, every=None, hertz=None, num_frames=None):
+        self.every = every
+        self.hertz = hertz
+        self.num_frames = num_frames
+
+    def _convert(self, video):
+        if not hasattr(video, "frame_index"):
+            frame_index = range(video.n_frames)
+        else:
+            frame_index = video.frame_index
+        
+        if not hasattr(video, "history"):
+            history = pd.DataFrame(columns=["filter", "value", "n_frames"])
+        else:
+            history = video.history
+
+        if self.every is not None:
+            new_idx = range(video.n_frames)[::self.every]
+            history.loc[history.shape[0]]= ["every", self.every, len(new_idx)]
```

Comment:
Suppose you run a `DerivedVideoStim` through the converter again... let's say the old `history` has 1000 rows,  selected with `every`, but now we're only grabbing `top_n=100`. Then I think `history` will only have 100 rows updated, but the last 900 will persist untouched, even though they're no longer actually contained in the stim. Is that right? If so, I think we'll need to either figure out a more systematic way to track history, or just not persist `history` across stims.

For now I think the latter is probably the way to go. I do think at some point we should implement more systematic logging/history capabilities, but maybe the place to do that properly is in the context of the graph module. I.e., rather than creating a detailed history that's bound to each stim, we can create a `GraphLog` that compactly tracks provenance properly. One way to do that would be to reconstruct history on-the-fly from the graph rather than storing it. E.g., if we know that a `VideoStim` was fed to a `FrameSamplingConverter` twice in a row, with different options, then we can reconstruct the full history from the `.history` attribute of each `DerivedVideoStim`. But this is a low priority and for now I vote for just tracking only the proximal history inside each `Stim`.